### PR TITLE
Added Bubble Sort in Java

### LINKED ISSUE #1363: Add Bubble Sort in Java
See requirements: https://sample-programs.therenegadecoder.com/projects/bubble-sort/
Add snippet here: https://github.com/TheRenegadeCoder/sample-programs/tree/master/archive/j/java

### PR DESCRIPTION
**Pull request by nickb82**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][1]! For simplicity, please make sure that your pull request includes one and only one contribution.

## Complete the Applicable Sections Below

Find which section best describes your pull request and make sure you fill it out. To start, let us know which issue you've fixed.

- [x] I fixed #1363

### Code Snippets

- [x] I named the pull request using `Added/Updated <Sample Program> in <Language>` format
- [ ] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])

## Notes
Created a bubble sort in java using other methods called copyArrayData() and addInt()

[1]: https://therenegadecoder.com/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[3]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/CODE_ARTICLE_TEMPLATE.md
[4]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/LANGUAGE_ARTICLE_TEMPLATE.md
[5]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/PROJECT_ARTICLE_TEMPLATE.md

### CODE REVIEW

**Comment by nickb82** [comment]
Sorry I am new to git and github but in ReadMe would you like me to update the text document with what I have done and the name of the file?

**Comment by jrg94** [comment]
Yeah, all you need to add is the name of the program to the list of programs at the top.

File: BubbleSortMainClass.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
+
+**
```
Comment (jrg94) [inline BubbleSortMainClass.java]: Can you move the main method to the other file, and delete this file? I'd like to keep BubbleSort as the original name.

**Comment by nickb82** [comment]
Hi, I was wondering if the changes I made were acceptable?

**Comment by nickb82** [comment]
How long does it take to test cause for me it looks like its in a queue waiting to be tested

**Comment by jrg94** [comment]
Oddly enough, it finished, but GitHub seems to be bugging out.

**Comment by nickb82** [comment]
Did I do this right?

$ git pull origin master
From https://github.com/nickb82/sample-programs
 * branch            master     -> FETCH_HEAD
Already up to date.

**Comment by nickb82** [comment]
I also tried this:

$ git fetch origin
remote: Enumerating objects: 20, done.
remote: Counting objects: 100% (20/20), done.
remote: Total 28 (delta 20), reused 20 (delta 20), pack-reused 8
Unpacking objects: 100% (28/28), done.
From https://github.com/nickb82/sample-programs
   95d6c99..161be12  bubble_sort_solution -> origin/bubble_sort_solution

then this:

$ git merge origin/master
Already up to date.

**Comment by jrg94** [comment]
So I'm not the biggest command line git expert. I usually lean on @auroq for that. But, if you use the GUI, you can usually do the following:

1. Switch to master and pull (this should update your local master branch)
2. Switch to this branch and merge master into it (ctrl + shift + u)

**Comment by nickb82** [comment]
Ok thank you, I believe I fixed it and updated local branch
.

**Comment by jrg94** [comment]
Make sure you push to the remote branch. That way, the changes will update here.

**Comment by nickb82** [comment]
Ok just pushed the local to master, did that work?

**Comment by jrg94** [comment]
I'm going to try making a pull request on your repo. Let's see if that works.

**Comment by nickb82** [comment]
Ok thank you

**Comment by nickb82** [comment]
I just merged and if you need me to do again please let me know

**Comment by nickb82** [comment]
Hi, I tried the steps you ask me to do and it just tells me in multiple steps that things are up to date. Did anything change or improve?

**Comment by jrg94** [comment]
Follow that link in the previous comment. I made a pull request on your fork.

**Comment by jrg94** [comment]
Yes! Your master branch should be updated now.

**Comment by nickb82** [comment]
Ok I followed the steps and I believe I did it correctly. Does everything look good on your end

**Comment by jrg94** [comment]
Everything seems to be good. Now, we're just waiting on testing.

**Comment by nickb82** [comment]
If I am understanding the conflict, it seems like you have your own main method now which I did not see before and that you are using ArrayList instead of an Array. Is this correct?

**Comment by jrg94** [comment]
Actually, I accidentally merged a solution to this already. Since we have worked so hard on this pull request, I don't necessarily want that to go to waste. Feel free to manipulate the solution to your tastes.

**Comment by nickb82** [comment]
I can make it however you want I just never really found any specifics on how to go about it, just a webpage that tells you what bubble sort does. If you want it in an ArrayList I can do that and if you also want the main to take user input I can. My solution just gives a bit more control via the main method by letting you add numbers into an array instead of asking a user for input. So what ever you need it to be I can adjust the code to that. Sorry for the miscommunication.

**Comment by jrg94** [comment]
We typically enforce whatever is in the testing table. In other words, the solution needs to work via the command line. As for specifics, we don't really care.

**Comment by nickb82** [comment]
Okay I believe I will have to use user input then to make it cleaner and easier for the command line. I will update and change the code for these specifications.

**Comment by jrg94** [comment]
Gonna close this out for the time being. We can try to piece something together for another solution soon! 👍

**Pull request closed without merging**